docs/content/guides/dagster-pipes/integrating-subprocess-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,244 @@
 ---
 title: "Integrating Subprocess with Dagster Pipes | Dagster Docs"
-description: ""
+description: "Learn how to use Dagster Pipes's built-in subprocess implementation to invoke a subprocess with the given command and environment."
 ---
 
 # Integrating Subprocess with Dagster Pipes
+
+In this guide, we’ll show you how to use [Dagster Pipes](/guides/dagster-pipes) with Dagster’s built-in subprocess implementation to run a subprocess with the given command and environment. You can then send information such as structured metadata and logging back to Dagster from the subprocess, where it will be visible in the Dagster UI.
+
+<Note>
+
+This guide focuses on using an out-of-the-box `PipesSubprocessClient` resource. For further customization with the subprocess invocation, use <PyObject module="dagster_pipes" object="open_dagster_pipes"/> approach instead. Refer to [Customizing Dagster Pipes protocols](/guides/dagster-pipes/customizing-dagster-pipes-protocols) for more info.
+
+</Note>
+
+---
+
+## Prerequisites
+
+To use Dagster Pipes to run a subprocess , you’ll need:
+
+- To have Dagster (`dagster`) and the Dagster UI (`dagster-webserver`) installed. Refer to the [Installation guide](/getting-started/install) for more info.
+
+---
+
+## Step 1: Create a Dagster asset that executes a subprocess
+
+In this step, you’ll create a Dagster asset that, when materialized, opens a Dagster pipes session and invokes subprocess that executes some external code.
```

Comment:
This reads awkwardly as "when materialized" could be interpreted as "when the external script has completed"

"asset that, in its execution function" could be good language.

Reply:
we refer to pipes process elsewhere, pipes session; what is the noun?